compiler/src/dotty/tools/dotc/quoted/PickledQuotes.scala

```diff
@@ -181,25 +181,38 @@ object PickledQuotes {
 
   /** Unpickle TASTY bytes into it's tree */
   private def unpickle(pickled: String | List[String], isType: Boolean)(using Context): Tree = {
-    val bytes = pickled match
-      case pickled: String => TastyString.unpickle(pickled)
-      case pickled: List[String] => TastyString.unpickle(pickled)
+    QuotesCache.getTree(pickled) match
+      case Some(tree) =>
+        quotePickling.println(s"**** Using cached quote for TASTY\n$tree")
+        treeOwner(tree) match
+          case Some(owner) =>
+            // Copy the cached tree to make sure the all definitions are unique.
+            TreeTypeMap(oldOwners = List(owner), newOwners = List(owner)).apply(tree)
+          case _ =>
+            tree
```

Comment:
In case we have optimizations that make the `TreeTypeMap` and identity function, do we have a test to guard against such optimizations?

Reply:
We already do that optimization outside `TreeTypeMap`, for example in `changeOwner`.

We do have tests that would fail if this stops coining the symbols.